### PR TITLE
[Bug 1173515] React powered single page AAQ  

### PR DESCRIPTION
**Pull request by mythmon**
This is an early WIP of the single page AAQ I've been working on. It is a separate view entirely from the previous AAQ. The goal is to do the entire AAQ flow from when the user clicks "ask a question" to when the user submits the question entirely on a single page.

This is structured using the Flux pattern. The major parts, following the pattern are:

* The Flux dispatcher, which is a singleton.
* A single store to keep track of the question being built up, as well as search suggestions.
* Action creators to manipulate the store(s).
* A controller-component: a React component that is the top level of the page, and interacts directly with the store(s).
* Other components which don't interact (directly) with the stores to build the page.

A rough idea of what is done and what is yet to do:

* [x] Bootstrap
* [x] Select product (old AAQ step 1)
* [x] Selected topic (old AAQ step 2)
* [ ] Suggestions before typing in a title based on the product and topic (old AAQ step 3)
* [x] Suggestions based on title (old AAQ step 4)
* [x] Entering title and content (old AAQ step 5)
* [ ] Log-in/Registration (old AAQ step 6)
* [x] Actually submitting the question via the API
* [ ] Automatically tagging questions
* [ ] Make it look better/be nicer
* [ ] Instructive text like "Be detailed, be nice"
* [ ] Attachments
* [ ] Advanced troubleshooting data (from the UITour library)

I'm doing two thing a bit odd here: one, I'm putting this PR up quite early, and two, the branch I'm using is on the Mozilla repo. This isn't to help with deploying to stage (though we will probably need to do that eventually). Instead the idea is that if anyone wants to contribute to this, they can push commits to this branch. Since this is quite a large project, that might make sense. Alternatively, I can just keep plugging away at this. The checkboxes above are editable by anyone with commit access. If you push commits that fix those things, please check them off.

@rehandalal You asked if I needed any help with this. If you find you have some extra time, I think I've figured out how you can help. Parts of this page would use much better styling. That doesn't involve touching Flux much, just editing the "html" in the components and the CSS. If you have the time and want to work on that (or anything else), I'd appreciate it.

**Screenshot of the current state** (click to embiggen)

<a href="https://cloud.githubusercontent.com/assets/305049/8146418/209d7c28-11eb-11e5-8f5a-9511cd976589.png"></a>

### CODE REVIEW

**Comment by safwanrahman** [comment]
@mythmon Can you please check the Bug number you mentioned in the PR? :confused:

**Comment by mythmon** [comment]
Oops, thanks. I typed a 7 to a 6. The bug number in the PR title is now correct. The branch has the wrong number though :(

**Comment by mythmon** [comment]
@rehandalal I think think this is ready to go. It isn't ready for prime time yet, but it is at a separate url, so it should be fine. Can you r? it, and then we can land it?

File: kitsune/bundles.py [hunk 1]
```diff
@@ -45,7 +45,7 @@
     # TODO: remove dependency on jquery ui CSS and use our own
     'jqueryui': {
         'source_filenames': (
-            'sumo/css/jqueryui/jqueryui.css',
+            'sumo/css/jqueryui/jqueryui.ess',
```
Comment (rehandalal) [inline kitsune/bundles.py]: did you mean to change this?

Comment (mythmon) [inline kitsune/bundles.py]: Oops, no.

File: kitsune/questions/static/questions/js/constants/AAQConstants.es6.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import constantMap from '../../../sumo/js/utils/constantMap.es6';
```
Comment (rehandalal) [inline kitsune/questions/static/questions/js/constants/AAQConstants.es6.js]: should this be .es6.js?

File: kitsune/questions/static/questions/js/stores/QuestionEditStore.es6.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
+/* globals _:false */
+
+import BaseStore from '../../../sumo/js/stores/BaseStore.es6';
+import Dispatcher from '../../../sumo/js/Dispatcher.es6';
+import {actionTypes, questionEditState} from '../constants/AAQConstants.es6';
```
Comment (rehandalal) [inline kitsune/questions/static/questions/js/stores/QuestionEditStore.es6.js]: same with these?

File: kitsune/sumo/static/sumo/js/utils/index.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+export apiFetch from './apiFetch.es6';
+export constantMap from './constantMap.es6';
+export scrollTo from './scrollTo.es6';
```
Comment (rehandalal) [inline kitsune/sumo/static/sumo/js/utils/index.js]: same?

File: kitsune/users/static/users/js/constants/UserAuthConstants.es6.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import constantMap from '../../../sumo/js/utils/constantMap.es6';
```
Comment (rehandalal) [inline kitsune/users/static/users/js/constants/UserAuthConstants.es6.js]: same here?

File: kitsune/users/static/users/js/stores/UserAuthStore.es6.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
+/* globals _:false */
+
+import BaseStore from '../../../sumo/js/stores/BaseStore.es6';
+import Dispatcher from '../../../sumo/js/Dispatcher.es6';
+import {actionTypes} from '../constants/UserAuthConstants.es6';
```
Comment (rehandalal) [inline kitsune/users/static/users/js/stores/UserAuthStore.es6.js]: and this?

File: kitsune/questions/static/questions/js/constants/AAQConstants.es6.js (hunk 2 above)
Comment (mythmon) [inline kitsune/questions/static/questions/js/constants/AAQConstants.es6.js]: I updated all the `*.es6` imports to be `*.es6.js`, to be more consistent.

**Comment by rehandalal** [comment]
seems fine to me! what's the worst that could happen!

r+

File: bower.json [hunk 7]
```diff
@@ -25,6 +26,8 @@
     "nwmatcher": "nwmatcher#1.3.4",
     "pikaday": "~1.3.2",
     "react": "#0.13.0",
-    "underscore": "underscore#1.8.2"
+    "underscore": "underscore#1.8.2",
+    "fetch": "#0.9.0",
+    "es6-promise": "~2.2.0"
```
Comment (gregglind) [inline bower.json]: (a babel step can remove this dep.)

Comment (mythmon) [inline bower.json]: For now not all of our code is going through Babel, so we still need this polyfill. If we ever get to 100% Babel compiled JS, I'll keep this in mind, thanks!

**Pull request merged**